shipment_advice_planner/models/stock_picking.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    warehouse_id = fields.Many2one(
```

Comment:
I don't like this module adds a warehouse field on the picking. Also the warehouse is usually given by the picking type, not the location

Reply:
Yes, but the method to retrieve the warehouse from the picking type is not correct as it retrieves the first warehouse of the company.

As the method is more correct on location_id, this is better

Reply:
The primary purpose of this field is to group pickings by warehouse and generate a shipment advice for each group, as the warehouse is required for the shipment advice model. Without it, running `read_group` by warehouse_id would be challenging.

I understand your reservations about adding the field to this module, and it would be more logical to create a separate module called `stock_picking_warehouse`.

I am fine with getting the warehouse_id from the picking_type

Reply:
> Yes, but the method to retrieve the warehouse from the picking type is not correct as it retrieves the first warehouse of the company.
> 
> As the method is more correct on location_id, this is better

No, on the picking type, the warehouse is explicit. It is set when the operation type is created from the warehouse and if you add new picking types, you will always set the right warehouse. On the location, the warehouse is not explicit and not visible.

Reply:
ok, thanks for the hint,

please check my last commit.